src/source/batchUpdateDocumentsFromFile.ts

```diff
@@ -0,0 +1,65 @@
+import {getAllJsonFilesFromEntries} from '../help/file';
+import {FileConsumer} from '../help/fileConsumer';
+import {
+  BatchUpdateDocuments,
+  BatchUpdateDocumentsFromFiles,
+  FailedUploadCallback,
+  FieldAnalyser,
+  SuccessfulUploadCallback,
+} from '../index';
+import {parseAndGetDocumentBuilderFromJSONDocument} from '../validation/parseFile';
+import {PushSource} from './push';
```

Comment:
let's just remove the circular dependency

Reply:
? You need it to ref' PushSource from this file.

If you want to remove it you'll need to rewire/refactor a lot more code.
You can replace it by an `import type` tho